### PR TITLE
install qemu on docker

### PR DESCRIPTION
**Pull request by ttnie**
Signed-off-by: nietingting <nietingting@xiaomi.com>

## Summary
install qemu on docker
## Impact
no
## Testing
test on local host

### CODE REVIEW

File: tools/ci/docker/linux/Dockerfile [hunk 1]
```diff
@@ -212,6 +212,20 @@ RUN cd renesas-tools/source/newlib && \
...
+###############################################################################
+# Build image for qemu
+###############################################################################
+#FROM qemu
+# Install dependence package
+RUN apt-get install -y libglib2.0-dev libpixman-1-dev ninja-build
+# Download qemu and install
+RUN mkdir qemu && \
+  #curl -s -L "https://download.qemu.org/qemu-6.1.1.tar.xz" && \
+  wget -c "https://download.qemu.org/qemu-6.1.1.tar.xz" && \
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:224]: why not use curl

File: tools/ci/docker/linux/Dockerfile [hunk 2]
```diff
@@ -212,6 +212,20 @@ RUN cd renesas-tools/source/newlib && \
...
 
+###############################################################################
+# Build image for qemu
+###############################################################################
+#FROM qemu
+# Install dependence package
+RUN apt-get install -y libglib2.0-dev libpixman-1-dev ninja-build
+# Download qemu and install
+RUN mkdir qemu && \
+  #curl -s -L "https://download.qemu.org/qemu-6.1.1.tar.xz" && \
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:223]: why not use the latest official release

File: tools/ci/docker/linux/Dockerfile [hunk 3]
```diff
@@ -212,6 +212,20 @@ RUN cd renesas-tools/source/newlib && \
 RUN cd /tools/renesas-tools/build/gcc && \
   make && make install
 
+###############################################################################
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:215]: move to "Base image" section

Comment (ttnie) [reply tools/ci/docker/linux/Dockerfile:215]: done

File: tools/ci/docker/linux/Dockerfile (hunk 2 above)
Comment (ttnie) [reply tools/ci/docker/linux/Dockerfile:223]: update to 7.1.0

File: tools/ci/docker/linux/Dockerfile (hunk 1 above)
Comment (ttnie) [reply tools/ci/docker/linux/Dockerfile:224]: update to curl

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@ttnie please rebase the patch, the ci break is fixed here: https://github.com/apache/incubator-nuttx/pull/7299

**Comment by ttnie** [comment]
> 

rebase the code and trigger ci again.

**Review by xiaoxiang781216 — APPROVED** [review]

File: tools/ci/docker/linux/Dockerfile [hunk 4]
```diff
@@ -78,6 +78,15 @@ FROM builder-base AS nuttx-toolchain-base
 RUN mkdir /tools
 WORKDIR /tools
 
+# Install dependence package
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:81]: move before line 71?

File: tools/ci/docker/linux/Dockerfile [hunk 5]
```diff
@@ -209,6 +218,7 @@ RUN cd renesas-tools/source/newlib && \
 RUN cd /tools/renesas-tools/build/gcc && \
   make && make install
 
+
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:221]: revert the change

Comment (ttnie) [reply tools/ci/docker/linux/Dockerfile]: done

File: tools/ci/docker/linux/Dockerfile [hunk 6]
```diff
@@ -78,6 +81,13 @@ FROM builder-base AS nuttx-toolchain-base
 RUN mkdir /tools
 WORKDIR /tools
 
+# Download and install qemu
+RUN mkdir qemu && \
+  curl -s -L "https://download.qemu.org/qemu-7.1.0.tar.xz" && \
```
Comment (pkarashchenko) [inline tools/ci/docker/linux/Dockerfile:86]:
```suggestion
  curl -s -L "https://download.qemu.org/qemu-7.1.0.tar.xz"  \
```

**Review by pkarashchenko — APPROVED** [review]

File: tools/ci/docker/linux/Dockerfile (hunk 6 above)
Comment (ttnie) [reply tools/ci/docker/linux/Dockerfile]: fix it

**Comment by pkarashchenko** [comment]
Something is still wrong

File: tools/ci/docker/linux/Dockerfile [hunk 7]
```diff
@@ -70,6 +70,9 @@ RUN mkdir -p $CARGO_HOME \
 
 CMD [ "/bin/bash" ]
 
+# Install dependence package
+RUN apt-get install -y libglib2.0-dev libpixman-1-dev ninja-build
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:74]: merge to line 28

File: tools/ci/docker/linux/Dockerfile [hunk 8]
```diff
@@ -78,6 +81,13 @@ FROM builder-base AS nuttx-toolchain-base
 RUN mkdir /tools
 WORKDIR /tools
 
+# Download and install qemu
+RUN mkdir qemu && \
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:85]: move after line 70

File: tools/ci/docker/linux/Dockerfile [hunk 9]
```diff
@@ -78,6 +81,13 @@ FROM builder-base AS nuttx-toolchain-base
...
+  curl -s -L "https://download.qemu.org/qemu-7.1.0.tar.xz" \
+  | tar -C qemu --strip-components 1 -xJ && \
+  cd qemu && ./configure && \
+  make && make install
```
Comment (xiaoxiang781216) [inline tools/ci/docker/linux/Dockerfile:89]: need add copy rule in the last section

**Comment by ttnie** [comment]
> Something is still wrong

**Comment by ttnie** [comment]
add --prefix as configure part, then copy to /tools/qemu/bin, and added to env PATH

**Comment by xiaoxiang781216** [comment]
@btashton the storage is still full:
https://github.com/apache/incubator-nuttx/actions/runs/3262925794/jobs/5360800214

**Comment by btashton** [comment]
Hmm seems like building qemu is taking a lot of disk space. Do we actually need it to be part of the build image, or should it be a different container? If it was we could just have another job. I'm not sure how qemu was going to be used in CI, but you could have the qemu-test job pull in the build artifacts from the build stage and then not need all the build utilities.

**Comment by btashton** [comment]
Another option is to just install the ubuntu packages for qemu. Is there a reason we need to be building it?

**Comment by ttnie** [comment]
> Another option is to just install the ubuntu packages for qemu. Is there a reason we need to be building it?

update to install qemu instead of building it.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]
